stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/GiphyMediaAttachmentView.kt

```diff
@@ -0,0 +1,212 @@
+package io.getstream.chat.android.ui.message.list.adapter.view.internal
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.annotation.Px
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import androidx.core.view.updateLayoutParams
+import com.getstream.sdk.chat.images.loadAndResize
+import com.getstream.sdk.chat.model.ModelType
+import com.getstream.sdk.chat.utils.extensions.imagePreviewUrl
+import com.google.android.material.shape.ShapeAppearanceModel
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
+import io.getstream.chat.android.ui.common.extensions.internal.createStreamThemeWrapper
+import io.getstream.chat.android.ui.common.extensions.internal.dpToPx
+import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
+import io.getstream.chat.android.ui.databinding.StreamUiGiphyMediaAttachmentViewBinding
+import io.getstream.chat.android.ui.message.list.adapter.view.GiphyMediaAttachmentViewStyle
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * View used to display Giphy images inside a shapeable container.
+ */
+public class GiphyMediaAttachmentView : ConstraintLayout {
+
+    /**
+     * Binding generated for [io.getstream.chat.android.ui.R.layout.stream_ui_giphy_media_attachment_view].
+     */
+    internal val binding: StreamUiGiphyMediaAttachmentViewBinding =
+        StreamUiGiphyMediaAttachmentViewBinding.inflate(streamThemeInflater, this, true)
+
+    /**
+     * Style applied to [GiphyMediaAttachmentView].
+     */
+    private lateinit var style: GiphyMediaAttachmentViewStyle
+
+    public constructor(context: Context) : this(context, null, 0)
+
+    public constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+
+    public constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context.createStreamThemeWrapper(),
+        attrs,
+        defStyleAttr
+    ) {
+        init(attrs)
+    }
+
+    private fun init(attrs: AttributeSet?) {
+        style = GiphyMediaAttachmentViewStyle(context, attrs)
+
+        binding.loadingProgressBar.indeterminateDrawable = style.progressIcon
+        binding.giphyLabel.setImageDrawable(style.giphyIcon)
+
+        binding.imageView.scaleType = style.scaleType
+        binding.imageView.setBackgroundColor(style.imageBackgroundColor)
+    }
+
+    /**
+     * Displays a Giphy inside its container. Depending on [GiphyMediaAttachmentViewStyle.isConstantSizeEnabled]
```

Comment:
We don't use `GiphyMediaAttachmentViewStyle.isConstantSizeEnabled` anymore, this kdoc should be updated